core/src/main/java/io/parsingdata/metal/data/selection/ByToken.java

```diff
@@ -78,5 +78,22 @@ private static ParseValueList getAllValuesRecursive(final ParseGraph graph, fina
         if (head.isGraph()) { return tailResults.add(getAllValuesRecursive(head.asGraph(), definition)); }
         return tailResults;
     }
+    
+    public static ParseItemList getAllRoots(final ParseGraph graph, final Token definition) {
+        checkNotNull(graph, "graph");
+        checkNotNull(definition, "definition");
+        return (graph.getDefinition() == definition && !graph.isEmpty() ? ParseItemList.create(graph) : ParseItemList.EMPTY)
```

Comment:
Additionally, it's factored slightly differently in the final commit to deal with an edge case (bonus: full coverage!)